src/azure-cli/azure/cli/command_modules/storage/_params.py

```diff
@@ -281,6 +281,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage account update', resource_type=ResourceType.MGMT_STORAGE) as c:
         t_tls_version = self.get_models('MinimumTlsVersion', resource_type=ResourceType.MGMT_STORAGE)
         c.register_common_storage_account_options()
+        c.argument('sku', arg_type=get_enum_type(t_sku_name),
+                   help='Note that the SKU name cannot be updated to Standard_ZRS, Premium_LRS or Premium_ZRS, '
+                   'nor can accounts of those SKU names be updated to any other value')
```

Comment:
The allowed values still contain Standard_ZRS, Premium_LRS and Premium_ZRS in help message. They should be removed to avoid confusion.

Reply:
Actually you are right. There are previous sku type because they are still in model definition as you can see in
https://github.com/Azure/azure-sdk-for-python/blob/09521dfb517e0859ec961cae006fb728d787b565/sdk/storage/azure-mgmt-storage/azure/mgmt/storage/v2021_01_01/models/_models_py3.py#L5088.

Here is to align with SDK, so I haven't removed them. But I will check with service team and try to fix from swagger definition. 
For CLI side, I want to keep current deign with SDK first and refine with next SDK generation for the fix.